mcorpc/authz_rego.go

```diff
@@ -0,0 +1,161 @@
+package mcorpc
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/choria-io/go-choria/choria"
+	"github.com/choria-io/go-config"
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/sirupsen/logrus"
+)
+
+type regoMatcher interface {
+	Set(caller string, actions string, facts string, classes string, groups map[string][]string)
+	SetFile(f string)
+}
+
+type regoPolicy struct {
+	cfg     *config.Config
+	req     *Request
+	agent   *Agent
+	log     *logrus.Entry
+	matcher regoMatcher
+	groups  map[string][]string
```

Comment:
this seems kind of unused at the moment? lots of setting it and passing it around but nothing being done with it?

Reply:
Copy/paste from the actionpolicy code - kinda don't know what to do with it, will probably delete it.

Need to figure out how to accommodate external data.